Refactor installation scripts

miniche opened the pull request:
Refactor Mac installation, with Debian installation model :
- Improved error checks during installation/uninstall
- Added new log functions (with a log file)
- Fixing some bugs

miniche committed:
[WIP] Refactor Mac installation

Using "displayAndExec" and "displayTitle" functions.
Fixing some bugs

Guimove commented:
![images](https://f.cloud.github.com/assets/1172067/56982/2ea5feca-5b37-11e2-939e-ae5c1b009ecd.jpeg)

Ninir commented on bash/macos.inc.sh:
```diff
@@ -219,6 +214,8 @@ function macos_install {
         return 1
     fi
 
+    displayTitle "Removing any existing installation"
```
If it's something running, we should add `...` to add more information

miniche committed:
[WIP] Working on installation improvements (OS X)

miniche committed:
Refactoring bash functions

Ninir commented on install.sh:
```diff
@@ -86,19 +89,13 @@ if [ "install" == ${ACTUAL_MODE} ]; then
-
-                sleep 3;
 
+                # Display the success text!
```
It's either "Displays" or "Displaying"

miniche committed:
English spelling corrections

miniche committed:
Some others spelling fixes

miniche committed:
New log bash functions

Now, CentralReport installer can write messages in the log files (/var/log/centralreport.log). Log functions have some log levels (info, error, etc…)

Ninir commented on bash/debian.inc.sh:
```diff
@@ -62,7 +62,7 @@ function debian_stop_cr {
 
     echo -e "\033[44m\033[1;37m"
-    displaytTitle "Lauching CentralReport configuration assistant..."
+    displayTitle "Lauching CentralReport configuration assistant..."
```
It's not assistant, it's wizard ;)

Ninir commented on bash/debian.inc.sh:
```diff
@@ -219,12 +219,12 @@ function debian_install {
     fi
 
-    displaytTitle "Starting installing thirparties software"
+    displayTitle "Starting installing thirparties software..."
```
`Installing third-party software...`

Ninir commented on bash/functions.inc.sh:
```diff
@@ -3,8 +3,9 @@
 # CentralReport Unix/Linux Indev version.
 # By careful! Don't use in production environment!
 
+# Gets actual OS (Linux distrib or Unix OS)
```
`Gets current OS`

Ninir commented on bash/functions.inc.sh:
```diff
@@ -24,48 +28,41 @@ function getPythonIsInstalled {
-  displayMessage "$*" >&2
-}
-
+# Displays an error message and exit the current function or program
```
Typo: `and exits [...]`

Ninir commented on bash/functions.inc.sh:
```diff
@@ -24,48 +28,41 @@ function getPythonIsInstalled {
+    exit ${exitcode}
 }
 
+# Displays the message with current statut (.../ERR/OK), while executing the command.
```
Typo: `[...] with the current statuS [...]`

Ninir commented on bash/log.inc.sh:
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# CentralReport Unix/Linux Indev version.
+# By careful! Don't use in production environment!
```
By? :P shouldn't it be `Be`?

Ninir commented on bash/log.inc.sh:
```diff
@@ -0,0 +1,37 @@
+ERROR_FILE="/var/log/centralreport.log"
+
+# Writes a message in the log file
+function writeLog() {
```
IMO we should name it `log` as a log always writes something (in the standard output or in a file)

Maybe just "logFile"

Ninir commented on bash/log.inc.sh:
```diff
@@ -0,0 +1,37 @@
+}
+
+# Writes a message on the standard output
+function writeConsole() {
```
Same thing here, `logConsole` seems better to me

Ninir commented on bash/log.inc.sh:
```diff
@@ -0,0 +1,37 @@
+}
+
+# Writes a title on the standard output and writes the message in the log file
+function writeTitle() {
```
`printTitle`?

Ninir commented on bash/macos.inc.sh:
```diff
@@ -13,23 +13,24 @@
-            displayError "Error on starting CentralReport (Error code : $?)"
-            return 1
+        if [ ${RETURN_CODE} -ne "0" ]; then
+            writeError "Error on starting CentralReport (Error code : ${RETURN_CODE})"
```
I would have wrote: `Error when starting CentralReport`, meaning using `when` more than `on`

Ninir commented on bash/macos.inc.sh:
```diff
@@ -67,8 +67,9 @@ function macos_stop_cr {
-    echo -e "\033[1;32mLauching CentralReport configuration assistant..."
-    echo -e "\033[0m"
+    writeConsole "\033[1;32m"
+    writeInfo "Lauching CentralReport configuration assistant..."
```
Typo: `assistant` -> wizard

Ninir commented on bash/macos.inc.sh:
```diff
@@ -85,60 +86,63 @@ function macos_config_assistant {
+            return ${RETURN_CODE}
         else
-            echo "Done!"
+            writeLog "Startup plist removed"
```
Sometimes you are using a `!` at the end when an even has finished, sometimes not, that's weird... :smile:

Ninir commented on bash/macos.inc.sh:
```diff
@@ -212,68 +214,80 @@ function macos_cp_startup_plist {
-    echo " ** Starting installing thirparties software ** "
-    echo " (Please consult http://github.com/miniche/CentralReport for licenses) "
-    echo " "
+    writeTitle "Installing thirparties software..."
```
Typo: `third-party`

Ninir commented on install.sh:
```diff
@@ -15,18 +16,19 @@ ACTUAL_MODE=install                         # Modes : install, check
+writeConsole "  -------------- CentralReport installer --------------"
+writeConsole "\033[0;44m"
+writeConsole "  Welcome! This script will install CentralReport on your host."
+writeConsole "  If you wants more details, please visit http://github.com/miniche/CentralReport."
```
Typo: `If you want`

Ninir commented on install.sh:
```diff
@@ -15,18 +16,19 @@ ACTUAL_MODE=install                         # Modes : install, check
+writeConsole "  Welcome! This script will install CentralReport on your host."
+writeConsole "  If you wants more details, please visit http://github.com/miniche/CentralReport."
+writeConsole " "
+writeConsole " During installation, we can ask an administrator password. It permit CentralReport "
```
`When installing CentralReport, we may ask for your password. It will allow CentralReport to write files and directories such as the project binaries, logs, etc.`

Ninir commented on install.sh:
```diff
@@ -67,43 +69,39 @@ if [ "install" == ${ACTUAL_MODE} ]; then
 
-                displayError "Error during CentralReport installation..."
-                displayError "CentralReport isn't installed on this host."
+                writeError "Error during CentralReport installation!"
```
`Something went wrong when installing CentralReport!`

Ninir commented on install.sh:
```diff
@@ -67,43 +69,39 @@ if [ "install" == ${ACTUAL_MODE} ]; then
 
             if [ ${CURRENT_OS} == ${OS_MAC} ]; then
-                echo "Ok, I continue. I will install CentralReport on a mac"
+                writeInfo "Ok, I continue. I will install CentralReport on a mac"
```
`Processing... CentralReport will be installed on this Mac.`

Ninir commented on install.sh:
```diff
@@ -67,43 +69,39 @@ if [ "install" == ${ACTUAL_MODE} ]; then
 
             elif [ ${CURRENT_OS} == ${OS_DEBIAN} ]; then
-                echo "Ok. I continue. I will install CentralReport on Debian"
+                writeInfo "Ok. I continue. I will install CentralReport on Debian"
```
Please see: https://github.com/miniche/CentralReport/pull/16/files#L5R72

miniche committed:
[WIP] First wave of fixes

miniche committed:
Updated : log function names

Ninir commented on bash/debian.inc.sh:
```diff
@@ -76,60 +80,73 @@ function debian_config_assistant {
-            displayError "Error on deleting CentralReport config file at ${CONFIG_FILE} (Error code : $?)"
-            return 1
+        if [ ${RETURN_CODE} -ne "0" ]; then
+            logError "Error on deleting CentralReport config file at ${CONFIG_FILE} (Error code : ${RETURN_CODE})"
```
Typo: `Error when deleting CentralReport config file [...]`

Ninir commented on bash/debian.inc.sh:
```diff
@@ -76,60 +80,73 @@ function debian_config_assistant {
 
 function debian_remove_config {
 
+    logFile "Removing CentralReport config file..."
```
Typo: `[...] config fileS`

miniche replied:
I have only one config file :wink:

Ninir commented on bash/debian.inc.sh:
```diff
@@ -76,60 +80,73 @@ function debian_config_assistant {
-        if [ $? -ne "0" ]; then
-            displayError "Error on deleting startup script at ${STARTUP_DEBIAN} (Error code : $?)"
+        if [ ${RETURN_CODE} -ne "0" ]; then
+            logError "Error on deleting startup script at ${STARTUP_DEBIAN} (Error code : ${RETURN_CODE})"
```
Typo: `Error when deleting [...]`

Ninir commented on bash/debian.inc.sh:
```diff
@@ -76,60 +80,73 @@ function debian_config_assistant {
         else
 
-        displayAndExec "Removing startup service" update-rc.d -f centralreport.sh remove
+            displayAndExec "Removing startup service" update-rc.d -f centralreport.sh remove
```
missing `...` :P

Ninir commented on bash/debian.inc.sh:
```diff
@@ -76,60 +80,73 @@ function debian_config_assistant {
-                displayError "Error on removing startup script with update-rc.d (Error code : $?)"
-                return 1
+            if [ ${RETURN_CODE} -ne "0" ]; then
+                logError "Error on removing startup script with update-rc.d (Error code : ${RETURN_CODE})"
```
Typo: `Error when removing [...]`

Ninir commented on bash/debian.inc.sh:
```diff
@@ -142,14 +159,15 @@ function debian_cp_bin {
 
     if [ $? -ne "0" ]; then
-          displayError "Error on creating CentralReport dir at ${INSTALL_DIR} (Error code : $?)"
+          logError "Error on creating CentralReport dir at ${INSTALL_DIR} (Error code : $?)"
```
Typo: `Error when removing [...]`

Ninir commented on bash/debian.inc.sh:
```diff
@@ -142,14 +159,15 @@ function debian_cp_bin {
-            displayError "Error on copying CentralReport bin files in ${PARENT_DIR} (Error code : $?)"
-            return 1
+        if [ ${RETURN_CODE} -ne "0" ]; then
+            logError "Error on copying CentralReport bin files in ${PARENT_DIR} (Error code : ${RETURN_CODE})"
```
Typo: `Error when copying [...]`

Ninir commented on bash/debian.inc.sh:
```diff
@@ -160,17 +178,20 @@ function debian_cp_startup_plist {
+    RETURN_CODE="$?"
+
+    if [ ${RETURN_CODE} -ne "0" ]; then
+      logError "Error on copying startup script at ${STARTUP_PLIST} (Error code : ${RETURN_CODE})"
```
Typo: `Error when copying [...]`

Ninir commented on bash/debian.inc.sh:
```diff
@@ -160,17 +178,20 @@ function debian_cp_startup_plist {
-          displayError "Error on registering startup script with update-rc.d (Error code : $?)"
-          return 1
+        if [ ${RETURN_CODE} -ne "0" ]; then
+          logError "Error on registering startup script with update-rc.d (Error code : ${RETURN_CODE})"
```
Typo: `Error when registering [...]`

Ninir commented on bash/macos.inc.sh:
```diff
@@ -85,60 +86,63 @@ function macos_config_assistant {
-            displayError "Error on deleting CentralReport config file at ${CONFIG_FILE} (Error code : $?)"
-            return 1
+        if [ ${RETURN_CODE} -ne "0" ]; then
+            logError "Error on deleting CentralReport config file at ${CONFIG_FILE} (Error code : ${RETURN_CODE})"
```
Typo: `Error when removing [...]`

Ninir commented on bash/macos.inc.sh:
```diff
@@ -85,60 +86,63 @@ function macos_config_assistant {
         if [ $? -ne "0" ]; then
-            displayError "Error on deleting startup plist file at ${STARTUP_PLIST} (Error code : $?)"
-            return 1
+            logError "Error on deleting startup plist file at ${STARTUP_PLIST} (Error code : ${RETURN_CODE})"
```
Typo: `Error when removing [...]`

Ninir commented on bash/macos.inc.sh:
```diff
@@ -155,29 +159,28 @@ function macos_remove_startup_plist {
-          displayError "Error on creating CentralReport dir at ${INSTALL_DIR} (Error code : $?)"
-          return 1
+    if [ ${RETURN_CODE} -ne "0" ]; then
+          displayError "Error on creating CentralReport dir at ${INSTALL_DIR} (Error code : ${RETURN_CODE})"
```
Typo: `Error when creating [...]`

Ninir commented on bash/macos.inc.sh:
```diff
@@ -155,29 +159,28 @@ function macos_remove_startup_plist {
-            displayError "Error on copying CentralReport bin files in ${PARENT_DIR} (Error code : $?)"
-            return 1
+        if [ ${RETURN_CODE} -ne "0" ]; then
+            displayError "Error on copying CentralReport bin files in ${PARENT_DIR} (Error code : ${RETURN_CODE})"
```
Typo: `Error when copying [...]`

Ninir commented on bash/macos.inc.sh:
```diff
@@ -186,14 +189,13 @@ function macos_cp_bin {
-      displayError "Error on copying startup plist at ${STARTUP_PLIST} (Error code : $?)"
-      return 1
+    if [ ${RETURN_CODE} -ne "0" ]; then
+      displayError "Error on copying startup plist at ${STARTUP_PLIST} (Error code : ${RETURN_CODE})"
```
Typo: `Error when copying [...]`

Ninir commented on bash/macos.inc.sh:
```diff
@@ -212,68 +214,80 @@ function macos_cp_startup_plist {
     sudo -v
     if [ $? -ne 0 ]; then
-        displayError "Impossible to use root privileges"
+        logError "Impossible to use root privileges!"
```
`Unable to use root privileges`

miniche committed:
Some minor fixes

miniche committed:
Spelling fixes

miniche commented:
Good for me :)
